activerecord/lib/active_record/table_metadata.rb

```diff
@@ -27,9 +27,9 @@ def associated_with?(table_name)
         reflection
       elsif ActiveRecord.allow_deprecated_singular_associations_name && reflection = klass&._reflect_on_association(table_name.singularize)
         ActiveSupport::Deprecation.warn(<<~MSG)
-          In Rails 7.2, referring to singular associations by their plural name will be deprecated.
-          To continue querying `#{table_name.singularize}` use `#{table_name}` instead.
-          You can get the new more performant behavior now by setting config.active_record.allow_deprecated_singular_associations_name = false
+          Referring to a singular association (e.g. `#{reflection.name}`) by its plural name (e.g. `#{reflection.plural_name}`) is deprecated.
```

Comment:
Should the `reflection.plural_name` here be the table name since that is what they passed in?

I am worried about a case where you pass in the plural table name and we instead say you can't pass the plural reflection name which seems confusing.

Reply:
Ah, I think if you did that **and** we found the reflection from the singular name, it should be the same unless it is a inflection difference, which I don't think we should need to worry about here.